### PR TITLE
feat(starfish): Switch all `sps()` measurements to `spm()`

### PR DESCRIPTION
**Pull request by gggritso**
This was a bit of a hike because we hard-coded the rate formatting in a bunch of places, and also duplicated a bunch of code, so I had to do some cleanup, too

The upshot is that "Spans per Second" is now "Spans per Minute" because for lower-throughput projects, they were just getting 0 everywhere which is a huge bummer. It's much easier to format a large number to be friendly than to show a very small number.

Screenshot 2023-07-25 at 3 00 41 PM
Screenshot 2023-07-25 at 3 00 44 PM
Screenshot 2023-07-25 at 3 00 46 PM

## Changes

- Allow passing a rate _unit_ through to formatters. Surprisingly there's not much precedent for this, even though even a `"number"` can have a unit
- Improve `formatAbbreviatedNumber` by adding a few specs, and enforcing consistent return type and formatting
- Change all references to `sps()` to `spm()`, and use abbreviated number formatting

### CODE REVIEW

File: static/app/utils/discover/fields.tsx [hunk 1]
```diff
@@ -118,6 +118,18 @@ export type Column = QueryFieldValue;
...
+export enum RateUnits {
+  PER_SECOND = '1/second',
+  PER_MINUTE = '1/minute',
+  PER_HOUR = '1/hour',
+}
+
+export const RATE_UNIT_LABELS = {
+  [RateUnits.PER_SECOND]: '/s',
+  [RateUnits.PER_MINUTE]: '/min',
+  [RateUnits.PER_HOUR]: '/hr',
```
Comment (AbhiPrasad) [inline static/app/utils/discover/fields.tsx:130]: l: As per the [non-SI units](https://en.wikipedia.org/wiki/Non-SI_units_mentioned_in_the_SI) still considered SI, hour is typically represented by `h`.

Do we want to switch to `/h` then?

File: static/app/views/starfish/views/webServiceView/starfishView.tsx [hunk 2]
```diff
@@ -123,12 +124,13 @@ export function StarfishView(props: BaseStarfishViewProps) {
                     bottom: '0',
                   }}
                   aggregateOutputFormat="rate"
+                  rateUnit={RateUnits.PER_SECOND}
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/webServiceView/starfishView.tsx:127]: l: it feels weird to me that we have to duplicate the `rateUnit` both here and within in `valueFormatter`. Something to improve for later on maybe.

File: static/app/utils/formatters.tsx [hunk 3]
```diff
@@ -308,6 +309,8 @@ export function formatAbbreviatedNumber(number: number | string) {
   return number.toLocaleString();
 }
 
-export function formatRate(value: number, rateUnit?: string) {
-  return `${value} /${rateUnit ?? 's'}`;
+export function formatRate(value: number, unit?: RateUnits) {
+  return `${formatAbbreviatedNumber(value)}${
+    RATE_UNIT_LABELS[unit ?? RateUnits.PER_SECOND]
+  }`;
```
Comment (AbhiPrasad) [inline static/app/utils/formatters.tsx:315]:
```suggestion
export function formatRate(value: number, unit: RateUnits = RateUnits.PER_SECOND) {
  return `${formatAbbreviatedNumber(value)}${
    RATE_UNIT_LABELS[unit]
  }`;
```

l: I think this signals intent a lot better than having the nullish coalescing operator

**Review by AbhiPrasad — APPROVED** [review]

File: static/app/utils/formatters.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/utils/formatters.tsx]: Agreed!

File: static/app/views/starfish/views/webServiceView/starfishView.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/starfish/views/webServiceView/starfishView.tsx:127]: Yep, for sure! I think the unit should be part of the `Series[]` to be honest, but I haven't looked into it deeply yet

File: static/app/utils/discover/fields.tsx (hunk 1 above)
Comment (gggritso) [reply static/app/utils/discover/fields.tsx:130]: Sold!

**Pull request merged**